sklearn/linear_model/_theil_sen.py

```diff
@@ -62,7 +62,8 @@ def _modified_weiszfeld_step(X, x_old):
 
     diff = diff[mask]
     diff_norm = diff_norm[mask][:, np.newaxis]
-    quotient_norm = linalg.norm(np.sum(diff / diff_norm, axis=0))
+    quotient_norm = linalg.norm(np.sum(diff / diff_norm, axis=0),
```

Comment:
The operation `np.sum(diff / diff_norm, axis=0)` can be nonfinite even if all entries was finite:

```python
from sklearn.utils.extmath import safe_sparse_dot
import numpy as np

f_max = np.finfo(np.float64).max

diff = np.full((2, 2), fill_value=f_max * 0.9)
diff_norm = 1

np.sum(diff / diff_norm, axis=0)
# array([inf, inf])
```